core/services/blockhashstore/feeder.go

```diff
@@ -0,0 +1,156 @@
+package blockhashstore
+
+import (
+	"context"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+)
+
+// Event contains metadata about a VRF randomness request or fulfillment.
+type Event struct {
+	// ID of the relevant VRF request.
+	ID string
+
+	// Block that the request or fulfillment was included in.
+	Block uint64
+}
+
+// Coordinator defines an interface for fetching request and fulfillment metadata from a VRF
+// coordinator.
+type Coordinator interface {
+	// Requests fetches VRF requests that occurred within the specified blocks.
+	Requests(ctx context.Context, fromBlock uint64, toBlock uint64) ([]Event, error)
+
+	// Fulfillments fetches VRF fulfillments that occured since the specified block.
+	Fulfillments(ctx context.Context, fromBlock uint64) ([]Event, error)
```

Comment:
In the Python BHS Feeder, we implicitly specify a `toBlock` for the fulfillment search. Any reason we don't do that here? Not that I think it's incorrect or anything, just curious.